include/reactor-cpp/multiport_callback.hh

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2022 TU Dresden
+ * All rights reserved.
+ *
+ * Authors:
+ *   Tassilo Tanneberger
+ */
+
+#ifndef REACTOR_CPP_MULTIPORT_CALLBACK_HH
+#define REACTOR_CPP_MULTIPORT_CALLBACK_HH
+
+#include <algorithm>
+#include <atomic>
+#include <iostream>
+#include <type_traits>
+#include <vector>
+
+namespace multiport {
+
+// fancy custom type_trait taken from stackoverflow
+// which checks if given class has the member function deactivate
+template <typename T> class has_deactivate {
```

Comment:
Since every `BasePort` has a `deactivate` method, why do we need to have this check? Will we ever put something else then ports in the Multiport?